src/components/ListItem.jsx

```diff
@@ -1,5 +1,76 @@
+import { useEffect, useState } from 'react';
+import { useToggle } from '@uidotdev/usehooks';
+import { Toggle } from './Toggle.jsx';
 import './ListItem.css';
+import { updateItem } from '../api/firebase.js';
+// import { getFutureDate} from '../utils/dates.js'
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({
+	name,
+	listPath,
+	itemId,
+	totalPurchases,
+	dateLastPurchased,
+}) {
+	const [purchased, setPurchased] = useToggle(false);
+	const [isDisabled, setIsDisabled] = useState(false);
+
+	useEffect(() => {
+		if (dateLastPurchased) {
+			const checkExpiration = () => {
+				const expirationDate = new Date(
+					dateLastPurchased.toMillis() + 24 * 60 * 60 * 1000,
+				);
+				const currentDate = new Date();
+				if (currentDate > expirationDate) {
+					setPurchased(false);
+					setIsDisabled(false);
+				} else {
+					setPurchased(true);
+					setIsDisabled(true);
+				}
+			};
+			checkExpiration();
+			const refreshInterval = setInterval(checkExpiration, 60000);
+			return () => clearInterval(refreshInterval);
+		}
+	}, [dateLastPurchased]);
+
+	const handleToggle = async () => {
+		const isPurchasing = !purchased;
+		setPurchased();
+
+		if (isPurchasing) {
+			try {
+				await updateItem(listPath, {
+					itemId,
+					totalPurchases: totalPurchases + 1,
+					dateLastPurchased: new Date(),
+				});
+				console.log(`${name} updated successfully`);
+				alert(`${name} is purchased successfully`);
+				setIsDisabled(true);
+			} catch (error) {
+				console.error('Error updating item:', error);
+				setPurchased(false);
+			}
+		}
+	};
+
+	return (
+		<div>
+			<li className="ListItem">
+				<div className="item-name">{name}</div>
+				<Toggle
+					toggle={handleToggle}
+					on={purchased}
+					name={name}
+					isDisabled={isDisabled}
+				/>
+				<div>
+					{dateLastPurchased ? dateLastPurchased.toDate().toLocaleString() : ''}
```

Comment:
I'm wondering if this dateLastPurchased variable could be passed into the Toggle component, to use as part of the label?

In the UI it looks like the checkbox label is all one group, e.g., "Purchased 9/7/2024, 3:51:13 AM". But by splitting it out into two separate divs, screen readers will announce each piece separately, e.g., "Purchased." (Move screen reader to the next element.) "9/7/2024, 3:51:13 AM."

Reply:
Thank you so much for the feedback, @meganesu! @shuveksha-tuladhar and I have decided to create an issue for this suggested change. You can find it under issue #9.1.

Thanks again for your suggestions!